bash-preexec.sh

```diff
@@ -47,7 +47,17 @@ __bp_last_argument_prev_command="$_"
 
 # Command to set our preexec trap. It's invoked once via
 # PROMPT_COMMAND and then removed.
-__bp_trap_install_string="trap '__bp_preexec_invoke_exec \"\$_\"' DEBUG;"
+__bp_trap_install() {
+  local prior_trap=$(trap -p DEBUG | sed "s/[^']*'\(.*\)'[^']*/\1/")
```

Comment:
Ok, this is fixed. It seems `set -T` is a non-starter, since it causes the DEBUG trap to be run from within command substitutions:

```
$ trap 'echo "from trap"' DEBUG
$ trap_cmd=$(trap -p DEBUG)
from trap
$ set -T
from trap
$ trap_t_cmd=$(trap -p DEBUG)
from trap
$ trap DEBUG
from trap
$ :
$ echo "$trap_cmd"
trap -- 'echo "from trap"' DEBUG
$ echo "$trap_t_cmd"
from trap
trap -- 'echo "from trap"' DEBUG
```

But the pattern I'm using now (capture the trap and clear it in `PROMPT_COMMAND`, then install the new trap in a function) appears to work.